## GENESIS API: Changes as of July 15, 2025

To improve the protection of your user data, the previously offered option of using the API via `GET` requests will be permanently deactivated on July 15, 2025. After this date, the GENESIS API will only be accessible using the `POST` method. The access data is sent in the header of the request and therefore cannot get into log files or be tapped during transport.
To help you with the changeover, we show executable code examples that use the POST method, starting with the **Python** programming language and the **Pandas** statistics package.

In [54]:
import requests
import pandas as pd
import io
import zipfile

We recommend using the **API token** to identify yourself to the web service. You will find this string in the [web service interface (API) menu](https://www-genesis.destatis.de/datenbank/online#modal=web-service-api) after logging in. It can be reset separately of your login data and therefore offers more control for shared projects, in training environments, etc. The token also does not contain any characters that may need to be masked and is therefore less prone to errors in use.
However, no writing access is possible after identification with a token. This includes changing the password via API or a table retrieval that is written to a queue or batch due to its size `(job=true)`. Such writing accesses must be identified by user name/email and password. The program examples described here can be used for both types of identification.
If you do not yet have access data for the GENESIS database, you can [register](https://www-genesis.destatis.de/datenbank/online#modal=register) easily and free of charge.

In [66]:
BASE_URL = 'https://www-genesis.destatis.de/genesisWS/rest/2020/'

TOKEN = "58b3_IHREN_TOKEN_EINFÜGEN_e6d032"

Regardless of the type of access data used, these must be transmitted in the request header of a POST request with the `‘Content-Type’: ‘application/x-www-form-urlencoded’`.

In [5]:
headers = {
    'Content-Type': 'application/x-www-form-urlencoded',
    'username': TOKEN,
    'password': ""
    }

langPref = "de"

## Research data cubes
The desired cube name required for the download can be researched using the `cubes` method via the `catalogue` service.  
In the following example, a list of cubes from the price statistics (`61*`) is requested.

In [41]:
responseCatalogue = requests.post(BASE_URL + 'catalogue/cubes',
    headers = headers,
    data = {
    'selection': '61*',
    'pagelength': 250,
    'language': langPref
    })

In [42]:
cubeList = responseCatalogue.json()["List"]
len(cubeList)

247

For clarity, we filter the list of 247 entries by searching for specific content

In [43]:
[x for x in cubeList if "Betriebsmittel" in x["Content"]]

[{'Code': '61221BJ001',
  'Content': 'Index der Einkaufspreise landwirt. Betriebsmittel, Index der Einkaufspreise landwirt. Betriebsmittel, Deutschland insgesamt, Landwirtschaftliche Betriebsmittel, Jahr',
  'State': 'vollständig mit Werten',
  'Time': '1961-2024',
  'LatestUpdate': '12.12.2024 08:01:02h',
  'Information': 'false'},
 {'Code': '61221BJ002',
  'Content': 'Index der Einkaufspreise landwirt. Betriebsmittel, Index der Einkaufspreise landwirt. Betriebsmittel, Deutschland insgesamt, Landwirtschaftliche Betriebsmittel, Wirtschaftsjahr',
  'State': 'vollständig mit Werten',
  'Time': '1938/39-2023/24',
  'LatestUpdate': '12.09.2024 08:01:01h',
  'Information': 'false'},
 {'Code': '61221BM001',
  'Content': 'Index der Einkaufspreise landwirt. Betriebsmittel, Index der Einkaufspreise landwirt. Betriebsmittel, Deutschland insgesamt, Landwirtschaftliche Betriebsmittel, Stichmonat',
  'State': 'vollständig mit Werten',
  'Time': '01/1968-01/2025',
  'LatestUpdate': '13.03.2025 18:05

### Download data cube

In [44]:
responseCube = requests.post(BASE_URL + 'data/cubefile',
    headers = headers,
    data = {
    'name': '61221BJ002',
    'language': langPref
    })

### Saving the file

In [46]:
with open("cube.txt", "wb") as f:
    f.write(responseCube.content)